### ISSUE TITLE
Make rev(sort(.)) -> sort(decreasing=TRUE) rule optional

### ISSUE THREAD

**Issue by MichaelChirico** [jimhester/lintr#3083]
Follow-up to #2614 / #3066.

Observe:

```r
x = c(a = 2, b = 2, c = 1)
rev(sort(x))
# b a c 
# 2 2 1 
sort(x, decreasing=TRUE)
# a b c 
# 2 2 1 
```

In code like `y <- rev(sort(x))[N]`, the difference can break back-compatibility. Therefore, I think we should make this lint optional, though I lean towards enabling it by default -- use cases for which this difference matters should probably be more intentional about their choice, including possibly marking `# nolint`. WDYT @Bisaloo / @IndrajeetPatil?

**Comment by Bisaloo** [comment]
Relying on a specific order for the ties seems extremely fragile and I would rather that users get a warning about it. The cost of adding `# nolint` is minimal for the rare cases when this may be what users want.
